### PR TITLE
add hyper link support, fix performance issue when measure the real width

### PR DESCRIPTION
**Pull request by ruiann**
use measureText instead of calculate each column

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,6 +86,14 @@ Danmaku can be emitted by calling `emit` method.
 damoo.emit({ text: "Damoo is awesome!", color: "#f49" });
 ```
 
+Add hyper link if you want to let the danmaku clickable
+
+加入超链接使弹幕能够点击跳转
+
+```javascript
+damoo.emit({ text: "Damoo is awesome!", color: "#f49", link : "script/Damoo.js"});
```
Comment (jamesliu96) [inline README.md:94]: it doesn't seemed to be working here. :disappointed:

File: src/damoo.js [hunk 2]
```diff
@@ -218,7 +243,11 @@
     };
 
     Thread.prototype.remove = function(d) {
-        var i = this.get(d).index;
+        var item = this.get(d);
+        if (item.link !== undefined) {
```
Comment (jamesliu96) [inline src/damoo.js:263]: I preferred `typeof val !== "undefined"` to detect undefined variables. :laughing:

File: src/damoo.js [hunk 3]
```diff
@@ -18,26 +18,44 @@
...
+        this.family = f;
     };
 
     var _preload = function(d, f) {
+        //create text canvas
         var cvs = Damoo.dom.createElement('canvas'),
             ctx = cvs.getContext('2d');
-        cvs.width = f.size * d.text.length * 1.2;
+        ctx.font = f.value;
+        cvs.width = ctx.measureText(d.text).width;
```
Comment (jamesliu96) [inline src/damoo.js:58]: this seems to be good, I'll take that. :bow:

File: README.md (hunk 1 above)
Comment (ruiann) [inline README.md:94]: I cannot get your point, maybe the hyper link is wrong? or you just don't want this function? Also I'm Chinese...

**Comment by jamesliu96** [comment]
@ruiann Can you remove the additional features but only leave the `measureText` part which I will take in, or maybe you may have to create a new pull request. Anyway, thank you very much! :smile:

**Pull request closed without merging**